deployment/serverless.yml

```diff
@@ -19,48 +19,91 @@ provider:
     Product: ${env:PROJECT_GROUP_SHORT}
     Programme: ${env:PROGRAMME}
     Service: ${env:PROJECT_GROUP_SHORT}
+  apiGateway:
+    apiKeySourceType: HEADER
+    apiKeys:
+      - name: ${env:NHS_UK_API_KEY_KEY}
+        value: ${self:custom.secret.${env:NHS_UK_API_KEY_KEY}}
+  endpointType: regional
+  tracing:
+    apiGateway: true
+    lambda: true
+
+custom:
+  secret: ${ssm:/aws/reference/secretsmanager/${env:API_GATEWAY_API_KEY_NAME}}
+
+  # iam:
+  #   role:
+  #     name: dos-integration-${sls:stage}-lambda-role
+  #     statements:
+        # - Effect: 'Allow'
+        #   Resource: 'TODO'
+        #   Action: 'TODO' #This need to be restricted when used
+        # - Effect: 'Deny'
+        #   Resource: 'TODO'
+        #   Action: 'TODO'
 
 plugins:
   - serverless-vpc-discovery
   - serverless-localstack
 
-custom:
-  vpcDiscovery:
-    # Documentation - https://www.serverless.com/plugins/serverless-vpc-discovery
-    vpcName: '${env:AWS_VPC_NAME}'
-    subnets:
-      - tagKey: Name
-        tagValues:
-          - '${env:AWS_VPC_NAME}-private-${env:AWS_REGION}a'
-          - '${env:AWS_VPC_NAME}-private-${env:AWS_REGION}b'
-          - '${env:AWS_VPC_NAME}-private-${env:AWS_REGION}c'
-    securityGroups:
-      - tagKey: Name
-        tagValues:
-          - '${env:TF_VAR_lambda_security_group_name}'
+# custom:
+#   vpcDiscovery:
+#     # Documentation - https://www.serverless.com/plugins/serverless-vpc-discovery
+#     vpcName: '${env:AWS_VPC_NAME}'
+#     subnets:
+#       - tagKey: Name
+#         tagValues:
+#           - '${env:AWS_VPC_NAME}-private-${env:AWS_REGION}a'
+#           - '${env:AWS_VPC_NAME}-private-${env:AWS_REGION}b'
+#           - '${env:AWS_VPC_NAME}-private-${env:AWS_REGION}c'
+#     securityGroups:
+#       - tagKey: Name
+#         tagValues:
+#           - 'uec-pu-apu-2429-rds'
 
 functions:
+  event-receiver:
+    image: ${env:AWS_SAME_ACCOUNT_DOCKER_REGISTRY}/event-receiver:${env:VERSION}
+    architecture: x86_64
+    name: ${env:PROJECT_ID}-${sls:stage}-event-receiver
+    description: ${sls:stage} Event Receiver lambda
+    memorySize: 512
+    timeout: 10
+    tags:
+      PYTHON_VERSION: ${env:PYTHON_VERSION}
+    environment:
+      PROFILE: ${env:PROFILE}
+      POWERTOOLS_SERVICE_NAME: ${env:TEAM_ID}
+      POWERTOOLS_TRACER_CAPTURE_RESPONSE: true
+      POWERTOOLS_TRACER_CAPTURE_ERROR: true
+      POWERTOOLS_TRACE_MIDDLEWARES: true
+      LOG_LEVEL: ${env:LOG_LEVEL}
+    events:
+      - http:
+          path: /dos-integration/api/v1/nhsuk-event-receiver/{type}
+          request:
+            parameters:
+              paths:
+                type: true # pharmacy /dental
+          method: post
+          private: true
   event-sender:
     image: ${env:AWS_SAME_ACCOUNT_DOCKER_REGISTRY}/event-sender:${env:VERSION}
     architecture: x86_64
     name: ${env:PROJECT_ID}-${sls:stage}-event-sender
     description: ${sls:stage} Event Sender lambda
     memorySize: 512
-    timeout: 60
```

Comment:
Can everything except tracing be re-added to this file

Reply:
Resolved